processors/base/helpers/__init__.py

```diff
@@ -266,3 +279,43 @@ def safe_prepend(prepend_string, string):
         string = '%s%s' % (prepend_string, string)
 
     return string
+
+
+def get_canonical_location_name(location):
+    """Find the canonical location name according to the
+    passed entry
+
+    Args:
+        location (str): the location to be normalized
+    """
+
+    clean_string = lambda u: sub(ur"\p{P}+", "", u).lower()
```

Comment:
Resolved. Your code couldn't be used since it's designed to string (I'm using unicode), but I've tried a similar approach without adding a new depedency